examples/cli/README.md

```diff
@@ -0,0 +1,122 @@
+# skorch helpers for command line interfaces (CLIs)
+
+Often you want to wrap up your experiments by writing a small script
+that allows others to reproduce your work. With the help of skorch and
+the fire library, it becomes very easy to write command line
+interfaces without boilerplate. All arguments pertaining to skorch or
+its PyTorch module are immediately available as command line
+arguments, without the need to write a custom parser. If docstrings in
+the numpydoc specification are available, there is also an
+comprehensive help for the user. Overall, this allows you to make your
+work reproducible without the usual hassle.
+
+This example is a showcase of how easy CLIs become with skorch.
+
+## Installation
+
+To use this functionaliy, you need some further libraries that are not
+part of skorch, namely fire and numpydoc. You can install them thusly:
+
+```bash
+pip install fire numpydoc
+```
+
+## Usage
+
+The `train.py` file contains an example of how to write your own CLI
+with the help of skorch. As you can see, this file almost exclusively
+consists of the proper logic, there is no argument parsing
+involved.
+
+When you write your own script, only the following bits need to be
+added:
+
+```python
+
+import fire
+from skorch.helper import parse_args
+
+# your model definition and data fetching code below
+...
+
+def main(**kwargs):
+    X, y = get_data()
+    my_model = get_model()
+
+    # important: wrap the model with the parsed arguments
+    parsed = parse_args(kwargs)
+    my_model = parsed(my_model)
+
+    my_model.fit(X, y)
+
+
+if __name__ == '__main__':
+    fire.Fire(main)
+
+```
+
+This even works if your neural net is part of an sklearn pipeline, in
+which case the help extends to all other estimators of your pipeline.
+
+For more information on how to use fire, follow [this
+link](https://github.com/google/python-fire).
+
+## Restrictions
+
+Almost all arguments should work out of the box. Therefore, you get
+command line arguments for the number of epochs, learning rate, batch
+size, etc. for free. Morevoer, you can access the module paremeters
```

Comment:
Same here. I wonder if this documentation duplication can be avoided somehow?

Reply:
Yes, I wondered about that too. I wanted to avoid sending readers this way and that way to collect all relevant information, hence the duplication. Any ideas?